kvbc/test/categorization/block_merkle_category_unit_test.cpp

```diff
@@ -257,6 +258,113 @@ TEST_F(block_merkle_category, updates_with_deleted_keys) {
   ASSERT_EQ(expected_v5, tagged_versions[4]);
 }
 
+TEST_F(block_merkle_category, stale_node_creation_and_deletion) {
+  // Create the first block and read its stale keys
+  auto update = BlockMerkleInput{{{key1, val1}, {key2, val2}, {key3, val3}, {key4, val4}, {key5, val5}}};
+  const auto block_out1 = add(1, std::move(update));
+  auto ser_stale = db->get(BLOCK_MERKLE_STALE_CF, serialize(TreeVersion{1}));
+  auto stale = StaleKeys{};
+  deserialize(*ser_stale, stale);
+
+  // There's no stale keys on the first block creation, since nothing existed before the first block.
+  ASSERT_EQ(0, stale.internal_keys.size());
+  ASSERT_EQ(0, stale.leaf_keys.size());
+
+  // Create the second block and read its stale keys
+  const auto block_out2 = add(2, BlockMerkleInput{{{key2, "new_val"s}}, {{key3, key5}}});
+  ser_stale = db->get(BLOCK_MERKLE_STALE_CF, serialize(TreeVersion{2}));
+  stale = StaleKeys{};
+  deserialize(*ser_stale, stale);
+
+  // Adding a new block adds stale internal keys, but no stale leaf keys, as blocks aren't deleted.
+  ASSERT_LT(0, stale.internal_keys.size());
+  ASSERT_EQ(0, stale.leaf_keys.size());
+
+  // There have been no pruned blocks yet.
+  ASSERT_EQ(0, cat.getLastDeletedTreeVersion());
+  ASSERT_EQ(2, cat.getLatestTreeVersion());
+
+  // Pruning the first block causes key2, key3, and key5 from version 1 to be deleted.
+  // Keys 1, 4, 5 are still active at version 1 and so remain in the database.
```

Comment:
I think this comment applies to key 1 and key 4 only.

Reply:
Good catch!